### PR TITLE
Features - Load State - [RET-379]

### PR DESCRIPTION
**Pull request by mhfen**
### What's this PR do?
Adds page loading state to API-dependent pages.

### Where should the reviewer start?
`PageLoader`

### Any background context you want to provide?
Good amount of changes across BodyData, ProductMatches, Sessions to make this work smoothly. Please check my work on the stuff you wrote so I'm not making any mistakes.

### Does this add new dependencies? Is there an installation procedure? (bundle, bower, npm, etc)
No.

### What are the relevant tickets?
https://mondorobot.atlassian.net/browse/RET-379

### Who should review this PR?
@cprater @mhfen

### CODE REVIEW

**Comment by mhfen** [comment]
@cprater

File: src/client/pages/product-matches/product-matches.jsx [hunk 1]
```diff
@@ -194,15 +200,34 @@ export default class Product extends ProductMatchesData {
     )
   }
 
+  removeLoader() {
```
Comment (cprater) [inline src/client/pages/product-matches/product-matches.jsx]: Not urgent, but would be nice if we could abstract this out somewhere

**Comment by cprater** [comment]
It would be nice to abstract out the removeLoader methods at some point. Good here 👍

**Comment by mhfen** [comment]
Abstracted!

**Pull request merged**